src/server.c

```diff
@@ -5229,23 +5256,38 @@ int redisFork(int purpose) {
         server.stat_fork_rate = (double) zmalloc_used_memory() * 1000000 / server.stat_fork_time / (1024*1024*1024); /* GB per second. */
         latencyAddSampleIfNeeded("fork",server.stat_fork_time/1000);
         if (childpid == -1) {
+            if (isMutualExChildType(purpose)) closeChildInfoPipe();
             return -1;
         }
+
+        /* The child_pid and child_type are only for mutual exclusive children.
+         * other child types should handle and store their pid's in dedicated variables.
+         *
+         * Today, we allows CHILD_TYPE_LDB to run in parallel with the other forks types:
+         * - it doesn't used for production, so it will not make the server to be less efficient
+         * - used for debugging, and we don't want to block it from running while other
+         *   forks are running (like RDB and AOF) */
+        if (isMutualExChildType(purpose)) {
```

Comment:
To me "ExclusiveChildType" is more clear than mutually exclusive.

Reply:
@yossigo the LDB children can run in parallel to these, so they're not totally exclusive, just mutually exclusive.
do you still think your name is better? (i can live with either, and a bit uncomfortable with both)

Reply:
Yes I'm aware of that, and not totally happy with "exclusive" but it seems seems less obscure to me.

Reply:
let's be verbose and use `isMutuallyExlusiveChildType`

Reply:
happy to see `isMutuallyExlusiveChildType`, i rather accept long name than unclear shorthand